src/ServiceControl/Operations/ImportFailedAudits.cs

```diff
@@ -0,0 +1,86 @@
+namespace ServiceControl.Operations
+{
+    using System;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Logging;
+    using Raven.Client;
+    using Raven.Client.Indexes;
+
+    public class ImportFailedAudits
+    {
+        public ImportFailedAudits(IDocumentStore store, AuditImporter auditImporter)
+        {
+            this.store = store;
+            this.auditImporter = auditImporter;
+        }
+
+        public Task Run(CancellationTokenSource tokenSource)
+        {
+            source = tokenSource;
+            return Task.Factory.StartNew(() => Run<FailedAuditImport, FailedAuditImportIndex>(source.Token));
+        }
+
+        void Run<T, I>(CancellationToken token) where I : AbstractIndexCreationTask, new()
+        {
+            var succeeded = 0;
+            var failed = 0;
+            using (var session = store.OpenSession())
+            {
+                var query = session.Query<T, I>();
+                using (var ie = session.Advanced.Stream(query))
+                {
+                    while (!token.IsCancellationRequested && ie.MoveNext())
+                    {
+                        FailedTransportMessage dto = ((dynamic)ie.Current.Document).Message;
+                        try
+                        {
+                            var transportMessage = new TransportMessage(dto.Id, dto.Headers)
```

Comment:
In v5 there are some properties of TransportMessage that are not linked to corresponding headers. ReplyToAddress is one. Recoverable is one as well. In most cases, if the message has a corresponding header then it gets read from there, otherwise it has some internal state variable. If these are not persisted carefully, we run the risk of losing data

Reply:
Check https://github.com/Particular/NServiceBus/blob/support-5.2/src/NServiceBus.Core/TransportMessage.cs for a complete implementation of TransportMessage. It looks like TimeToBeReceived is one (and I was wrong about ReplyToAddress).

Reply:
@mikeminutillo what should we do with this property then? If set, assign it to a header?

Reply:
Can do, but then you need to set it on TransportMessage on the other end too. Recoverable is literally just a boolean on the class. I guess all I'm saying is that in V5, a transport message is slightly more than (id, body, headers) and we need to be sure we don't lose those extra bits

Reply:
Looking at the code only `Recoverable` and `TTBR` are not stored as headers. All the other stuff is. And for audits we don't really need these two as `Recoverable` is always set to `true` (for some legacy reason) and `TTBR` is not used.

Why do you think it is broken for recoverability?

Reply:
If the concern is around recoverability, it shouldn't actually be a concern since this is auditing? Also, since it's copy-paste behaviour from existing import pipeline it should be ok.

These are read-only once imported, right? ServiceInsight can't re-send audit messages, so even if we get the TTBR or Recoverable flags wrong, it's not a deal-breaker surely.

Reply:
OK, you have convinced me